--- dhcpcd-ui-0.5.1/src/dhcpcd-gtk/main.c~	2010-06-09 15:55:38.000000000 +0200
+++ dhcpcd-ui-0.5.1/src/dhcpcd-gtk/main.c	2011-04-07 11:43:05.405993616 +0200
@@ -194,11 +194,15 @@
 	g_strfreev(msgs);
 	if (nn != NULL)
 		notify_notification_close(nn, NULL);
+#if !defined(NOTIFY_VERSION_MINOR) || (NOTIFY_VERSION_MAJOR == 0 && NOTIFY_VERSION_MINOR < 7)
 	if (gtk_status_icon_get_visible(status_icon))
 		nn = notify_notification_new_with_status_icon(title,
 		    msg, icon, status_icon);
 	else
 		nn = notify_notification_new(title, msg, icon, NULL);
+#else
+	nn = notify_notification_new(title, msg, icon);
+#endif
 	notify_notification_set_timeout(nn, 5000);
 	g_signal_connect(nn, "closed", G_CALLBACK(notify_closed), NULL);
 	notify_notification_show(nn, NULL);
